github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml

```diff
@@ -637,6 +637,106 @@ postsubmits:
+    - name: post-project-infra-prow-performance-cluster-prometheus-stack
+      always_run: false
+      annotations:
+        testgrid-create-test-group: "false"
+      decorate: true
+      branches:
+      # regex for semver from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+      - ^prow-performance-workload-cluster-prometheus-stack-deployment-v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+      labels:
+        preset-docker-mirror-proxy: "true"
+      cluster: ibm-prow-jobs
+      spec:
+        securityContext:
+          runAsUser: 0
+        containers:
+        - image: quay.io/kubevirtci/bootstrap:v20210906-994b913
+          env:
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: /etc/gcs/service-account.json
+          - name: GIT_ASKPASS
+            value: "/home/prow/go/src/github.com/kubevirt/project-infra/hack/git-askpass.sh"
+          command:
+            - "/usr/local/bin/runner.sh"
+            - "/bin/bash"
+            - "-c"
+            - |
+              # install yq to decrypt_secrets
+              curl -Lo ./yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
+              chmod +x ./yq && mv ./yq /usr/local/bin/yq
+
+              # decrypt kubeconfig and grafana credentials
+              source ./hack/manage-secrets.sh
+              decrypt_secrets
+              extract_secret 'kubeconfigPerf' ~/.kube/config
+              mkdir -p /tmp/grafana/credentials
+              extract_secret 'grafanaUserPerf' /tmp/grafana/credentials/admin-user
+              extract_secret 'grafanaPasswordPerf' /tmp/grafana/credentials/admin-password
+
+              # prepare Kubevirt control plane custom dashboard
+              mkdir -p /tmp/grafana/custom-dashboards/
+              curl -Lo /tmp/grafana/custom-dashboards/kubevirt-control-plane.json https://raw.githubusercontent.com/kubevirt/monitoring/main/dashboards/grafana/kubevirt-control-plane.json
+              kubectl create ns monitoring
+              kubectl create -n monitoring configmap kubevirt-dashboards --from-file=/tmp/grafana/custom-dashboards/
+              kubectl label cm -n monitoring kubevirt-dashboards grafana_dashboard=true
+
+              # deploy a storage class to use with prometheus, so the pod restart doesn't lose everything
+              kubectl apply -f https://github.com/rancher/local-path-provisioner/raw/master/deploy/local-path-storage.yaml
+
+              # install helm to deploy prometheus operator
+              curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
+              chmod 700 get_helm.sh
+              ./get_helm.sh
+
+              helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+              helm install \
```

Comment:
Why introduce a new tool? We don't use helm in any of our other deployments. 

As I see it it would be a good idea to align with what we already have using bazel's rules_gitops, including tests, making it super easy to integrate with thanos and our alerting pipeline.

For instance this is the deployment job of the monitoring stack https://github.com/kubevirt/project-infra/blob/main/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml#L795 we would only need to add a new line for deploying to the new environment and entries for the new cluster here https://github.com/kubevirt/project-infra/tree/main/github/ci/services/prometheus-stack let me know wdyt

Reply:
Right, it might be better to make the prometheus deployment consistent with the other clusters, especially since you have some testing...

The idea of using helm was because it is very easy to update the prometheus-stack versions....

My only concern is with respect to grafana dashboards, I was deploying all the default ones plus the kubevirt one control plane dashboard. So, we might need some extra configuration besides adding a new line in `post-project-infra-prometheus-stack-deployment`

Reply:
Sure, you should add entries for the new cluster here https://github.com/kubevirt/project-infra/tree/main/github/ci/services/prometheus-stack for instance, we have different manifests for each of the environments and a common set of them here https://github.com/kubevirt/project-infra/tree/main/github/ci/services/prometheus-stack/manifests/stack 

You can extract the `grafana-dashboards-default` ConfigMap from https://github.com/kubevirt/project-infra/blob/main/github/ci/services/prometheus-stack/manifests/stack/production-control-plane/grafana.yaml#L180, leave the rest common to `production-control-plane` and `production-e2e-performance-workloads` and have different config maps with the actual dashboard contents for each of them

Reply:
Unfortunately we cannot use the https://github.com/kubevirt/project-infra/blob/main/github/ci/services/prometheus-stack/manifests/stack/common/kube-prometheus-stack.yaml

KubeVirt only exposes the metrics if prometheus operator is in the `monitoring` namespace, and the ServiceAccount is `prometheus-k8s`.....

So, we can change the current `kube-prometheus-stack.yaml` or create a new one for the performance cluster...

Reply:
No worries, rules_gitops takes care of moving all the resources to the namespace defined in the build file https://github.com/kubevirt/project-infra/blob/main/github/ci/services/prometheus-stack/BUILD.bazel#L40

In any case of any conflict we can always refactor and leave only the manifests required for each environment, but the defined namespace should not be a problem.